core/src/style/style.h

```diff
@@ -81,31 +63,42 @@ class Style {
     /* Draw mode to pass into <VboMesh>es created with this style */
     GLenum m_drawMode;
 
-    /* Set of strings defining which data layers this style applies to, 
-     * along with the style parameters corresponding to these data layers */
-    std::vector< std::pair<std::string, StyleParams> > m_layers;
-    
+    /* Set of strings defining which data layers this style applies to,
+     * along with the style paramter map corresponding to these data layers,
+     * to be parsed explicitly by styles for their style parameters*/
+    std::vector< std::pair<std::string, StyleParamMap> > m_layers;
+
+    // cache the styleParameters for a layer (later for a ruleID(uniquely generated))
+    // NOTE: no use at present!!!
+    std::unordered_map<std::string, void*> m_styleParamCache;
```

Comment:
Might make sense to make one of these for each specific style class and store the actual type of the parsed style parameters (instead of `void*`), then this container would take care of destroying all of those cached parameters when it destructs.

Thinking further down the road, we should be aware that there is also a drawback to caching these in the style classes: we'll need to ensure that this shared cache is threadsafe :\